hudi-common/src/main/java/org/apache/hudi/common/table/HoodieTimeline.java

```diff
@@ -250,7 +253,10 @@ static HoodieInstant getCompactionInflightInstant(final String timestamp) {
     return new HoodieInstant(State.INFLIGHT, COMPACTION_ACTION, timestamp);
   }
 
-  static HoodieInstant getInflightInstant(final HoodieInstant instant) {
+  static HoodieInstant getInflightInstant(final HoodieInstant instant, final HoodieTableType tableType) {
+    if ((tableType == HoodieTableType.MERGE_ON_READ) && instant.getAction().equals(COMMIT_ACTION)) {
```

Comment:
add some comments for this special handling? I 'd prefer keeping this here compared to pushing this to HoodieInstant

Reply:
Added